Loop/Managers/ResetLoopManager.swift

```diff
@@ -0,0 +1,120 @@
+//
+//  ResetLoopManager.swift
+//  Loop
+//
+//  Created by Cameron Ingham on 5/18/23.
+//  Copyright © 2023 LoopKit Authors. All rights reserved.
+//
+
+import LoopKit
+
+protocol ResetLoopManagerDelegate: AnyObject {
+    func loopWillReset()
+    func loopDidReset()
+    
+    func presentConfirmationAlert(
+        confirmAction: @escaping (_ pumpManager: PumpManager?, _ completion: @escaping () -> Void) -> Void,
+        cancelAction: @escaping () -> Void
+    )
+    
+    func presentCouldNotResetLoopAlert(error: Error)
+}
+
+class ResetLoopManager {
+    
+    private weak var delegate: ResetLoopManagerDelegate?
+    
+    private var loopIsAlreadyReset: Bool = false
+    private var resetAlertPresented: Bool = false
+    
+    init(delegate: ResetLoopManagerDelegate?) {
+        self.delegate = delegate
+        
+        checkIfLoopIsAlreadyReset()
+    }
+    
+    func askUserToConfirmLoopReset() {
+        if loopIsAlreadyReset {
+            UserDefaults.appGroup?.userRequestedLoopReset = false
+        }
+        
+        if UserDefaults.appGroup?.userRequestedLoopReset == true && !resetAlertPresented {
+            resetAlertPresented = true
+            
+            delegate?.presentConfirmationAlert(
+                confirmAction: { [weak self] pumpManager, completion in
+                    self?.resetAlertPresented = false
+                    
+                    guard let pumpManager else {
+                        self?.resetLoop()
+                        completion()
+                        return
+                    }
+                    
+                    pumpManager.prepareForDeactivation() { [weak self] error in
+                        guard let error = error else {
+                            self?.resetLoop()
+                            completion()
+                            return
+                        }
+                        
+                        self?.delegate?.presentCouldNotResetLoopAlert(error: error)
+                    }
+                }, cancelAction: { [weak self] in
+                    self?.resetAlertPresented = false
+                    UserDefaults.appGroup?.userRequestedLoopReset = false
+                }
+            )
+        }
+        
+        checkIfLoopIsAlreadyReset()
+    }
+    
+    private func checkIfLoopIsAlreadyReset() {
+        let fileManager = FileManager.default
+        
+        guard let url = fileManager.urls(for: .documentDirectory, in: .userDomainMask).first else {
+            return
+        }
+        
+        guard let hasReset = try? fileManager.contentsOfDirectory(atPath: url.path).count <= 1 else {
```

Comment:
Curious what the 1 items is?

Reply:
<img width="1163" alt="Screenshot 2023-05-19 at 10 37 41 AM" src="https://github.com/tidepool-org/Loop/assets/2769158/44c821fd-5f0c-462f-809f-b3c1de6089b9">

`protection.test` is created between `checkProtectedDataAvailable()` and `launchManagers()` where this check is made.